### PR TITLE
Feat: Trip title / description character limit

### PR DESCRIPTION
**Pull request by samueljklee**
[Trip title / description character limit](https://github.com/binhonglee/GlobeTrotte/wiki/Roadmap)

As discussed with @binhonglee , Title limit 50, Description limit 500

### CODE REVIEW

File: src/cockpit/tests/components/CEditItem.test.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import CEditItem from "@/components/CEditItem.vue";
+import Item from "@/shared/TripEditable";
+import { mockTripObj } from "@/tests/mockData/data";
+import { expect } from "@jest/globals";
+
+import { mount } from "@vue/test-utils";
+
+const mockedTrip = new mockTripObj();
```
Comment (binhonglee) [inline src/cockpit/tests/components/CEditItem.test.ts:8]: Remove this? (and the import above) since its not used.

**Review by binhonglee — CHANGES_REQUESTED** [review]
Seems like this is only to limit rendering of characters when user is editing the trip? Can we also add these limitation to `save()` in `CEditTrip()` so it shows an alert when users trying to save trip title / description over the limit?

**Comment by binhonglee** [comment]
Code pointer: https://github.com/binhonglee/GlobeTrotte/blob/main/src/cockpit/components/CEditTrip.vue#L106-L157

You can see how it shows an alert [here](https://github.com/binhonglee/GlobeTrotte/blob/main/src/cockpit/components/CEditTrip.vue#L133-L143) when the place input does not fit the requirement.

File: src/cockpit/components/CEditTrip.vue [hunk 2]
```diff
@@ -228,6 +242,31 @@ export default defineComponent({
...
+        this.showAlert(`Trip name is too short.`);
+        return false;
+      }
+      if (name.length > NAME_CHAR_MAX_COUNT) {
+        this.showAlert(
+          `Trip name cannot be longer than ${NAME_CHAR_MAX_COUNT}.`,
+        );
+        return false;
+      }
+      if (description.length > DESCRIPTION_CHAR_MAX_COUNT) {
```
Comment (binhonglee) [inline src/cockpit/components/CEditTrip.vue:257]: Do we want to add `!description` check as well?

File: src/cockpit/components/CEditTrip.vue [hunk 3]
```diff
@@ -228,6 +242,31 @@ export default defineComponent({
         E.get(E.get(this, "name"), "input").focus();
       });
     },
+    checkValidNameDescription(name: string, description: string): boolean {
+      console.table([name, description]);
```
Comment (binhonglee) [inline src/cockpit/components/CEditTrip.vue:246]: remove this?

File: src/cockpit/components/CEditItem.vue [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
 <script lang="ts">
 import { defineComponent } from "vue";
+import {
+  DESCRIPTION_CHAR_MAX_COUNT,
+  NAME_CHAR_MAX_COUNT,
+} from "@/shared/constants";
+
+/** Constants */
+const NAME_ROW_COUNT = 1;
+const DESCRIPTION_ROW_COUNT = 3;
```
Comment (binhonglee) [inline src/cockpit/components/CEditItem.vue:26]: Can we move this to `CEditTrip` as well? `CEditItem` is supposed to be pretty general and is currently used in many places including registration and login. We might not want to have the same restrictions for them.

File: src/cockpit/components/CEditItem.vue [hunk 5]
```diff
@@ -38,6 +47,14 @@ export default defineComponent({
...
       default: "",
     },
+    nameCharMaxCount: {
+      type: Number,
+      default: NAME_CHAR_MAX_COUNT,
+    },
+    descriptionCharMaxCount: {
+      type: Number,
+      default: DESCRIPTION_CHAR_MAX_COUNT,
+    },
```
Comment (binhonglee) [inline src/cockpit/components/CEditItem.vue:57]: We can turn this into a more generic prop that can be used by other UI components. (say something like minimum password length) and have it as empty by default.

**Review by binhonglee — COMMENTED** [review]
Do you think you can add some simple `CEditTrip` test as well? Current `CEditTrip` test is broken and can be deleted tbh.

LGTM otherwise

Let me know again when you're ready to merge and I'll approve it (mergify will auto-merge once I approve lol).

File: src/cockpit/components/CEditItem.vue (hunk 5 above)
Comment (samueljklee) [reply src/cockpit/components/CEditItem.vue:57]: What about `itemCharMaxCount`? We let parent component to specifically pass the max count? What's a default would be good? 50 char?

File: src/cockpit/components/CEditItem.vue (hunk 4 above)
Comment (samueljklee) [reply src/cockpit/components/CEditItem.vue:26]: We'll have this as props as well? And we'll have default as 1 row. How about that?

File: src/cockpit/components/CEditItem.vue (hunk 5 above)
Comment (binhonglee) [reply src/cockpit/components/CEditItem.vue:57]: Yea that sounds good.

File: src/cockpit/components/CEditItem.vue (hunk 4 above)
Comment (binhonglee) [reply src/cockpit/components/CEditItem.vue:26]: Yea I think that was the behavior before. Its 1 by default unless `$this.props.type` is specified as `textarea`.

File: src/cockpit/components/CEditTrip.vue (hunk 2 above)
Comment (samueljklee) [reply src/cockpit/components/CEditTrip.vue]: added

File: src/cockpit/components/CEditItem.vue (hunk 4 above)
Comment (samueljklee) [reply src/cockpit/components/CEditItem.vue]: refactored

File: src/cockpit/components/CEditItem.vue (hunk 5 above)
Comment (samueljklee) [reply src/cockpit/components/CEditItem.vue]: refactored

**Review by binhonglee — APPROVED** [review]

File: src/cockpit/tests/components/CEditTrip.test.ts [hunk 6]
```diff
@@ -4,18 +4,37 @@
...
-    },
-    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-  } as any);
-  t.regex(wrapper.find(".editLabel").text(), /City:/);
+describe("CEditTrip", () => {
+  it("renders component", () => {
+    const trip = mockedTrip.trip;
+    const wrapper = mount(CEditTrip, {
+      props: {
+        trip,
```
Comment (binhonglee) [inline src/cockpit/tests/components/CEditTrip.test.ts:17]: `isNew` is a required param that caused the typecheck error.

```suggestion
        trip: trip,
        isNew: false,
```

**Comment by binhonglee** [comment]
Oh shoot. I didn't update the mergify rule since I split gh-action into multi parts. Merging this manually instead.

**Pull request merged**